src/main/java/uk/gov/hmcts/reform/blobrouter/util/StorageClientManager.java

```diff
@@ -0,0 +1,44 @@
+package uk.gov.hmcts.reform.blobrouter.util;
+
+import com.azure.storage.blob.BlobServiceAsyncClient;
+import com.azure.storage.blob.models.BlobContainerItem;
+import org.slf4j.Logger;
+import reactor.core.publisher.Flux;
+import uk.gov.hmcts.reform.blobrouter.config.ServiceConfiguration;
+
+import java.util.Map;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+public final class StorageClientManager {
+
+    private static final Logger LOGGER = getLogger(StorageClientManager.class);
+
+    private static final String REJECTED_CONTAINER_SUFFIX = "rejected";
+
+    private StorageClientManager() {
+        // utility class constructor
+    }
+
+    public static Flux<BlobContainerItem> getAvailableContainers(
```

Comment:
For better testability, could this method be non-static?

Reply:
@mdayican your argument?

Reply:
if you are telling this for mocking we can still mock static methods.

Reply:
Mocking static methods is not quite as straightforward as instance mocks and has its own limitations. There's no cost in having instance methods.